docs/book/user-guide/production-guide/ci-cd.md

```diff
@@ -6,28 +6,135 @@ description: >-
+...
+```
+
+## Set up your secrets in Github
+
+For our Github Actions we will need to set up some
+secrets [for our repository](https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions#creating-secrets-for-a-repository).
+Specifically, you should use github secrets to store the `ZENML_API_KEY` that you created above.
+
+![create_gh_secret.png](../../.gitbook/assets/create_gh_secret.png)
+
+The other values that are loaded from secrets into the
+environment [here](https://github.com/zenml-io/zenml-gitflow/blob/main/.github/workflows/pipeline_run.yaml#L14-L23)
+can also be set explicitly or as variables.
+
+## Set up different stacks for Staging and Production
+
+You might not necessarily want to use the same stack with the same resources for your staging and production use.
+
+This step is optional, all you'll need for certain is a stack that runs remotely (remote orchestration and artifact
+storage). The rest is up to you. You might for example want to parametrize your pipeline to use different data sources
+for the respective environments. You can also use different [configuration files](../../how-to/configuring-zenml/configuring-zenml.md)
+for the different environments to configure the [Model](../../how-to/use-the-model-control-plane/README.md), the 
+[DockerSettings](../../how-to/customize-docker-builds/docker-settings-on-a-pipeline.md), the [ResourceSettings like
+accelerators](../../how-to/training-with-gpus/training-with-gpus.md) differently for the different environments.
+
+## Run Pipeline for PRs
+
+One way to ensure only fully working code makes it into production, you should use a staging environment to test all
+the changes made to your code base and verify they work as intended. To do so automatically you should set up a
+github action workflow that runs your pipeline for you when you make changes to it.
+[Here](https://github.com/zenml-io/zenml-gitflow/blob/main/.github/workflows/pipeline_run.yaml) is an example that you
+can use.
+
+To only run the Github Action on a PR, you can configure the yaml like this
+
+```yaml
+on:
+  pull_request:
+    branches: [ staging, main ]
+```
+
+When the workflow starts we want to set some important values. Here is a simplified version that you can use.
+
+```yaml
+jobs:
+  run-staging-workflow:
+    runs-on: run-zenml-pipeline
+    env:
+      ZENML_HOST: ${{ secrets.ZENML_HOST }}  # Put your server url here
+      ZENML_API_KEY: ${{ secrets.ZENML_API_KEY }}  # Retrieves the api key for use  
+      ZENML_STACK: stack_name  #  Use this to decide which stack is used for staging
+      ZENML_GITHUB_SHA: ${{ github.event.pull_request.head.sha }}
+      ZENML_GITHUB_URL_PR: ${{ github.event.pull_request._links.html.href }}
+```
```

Comment:
you can link to the full github actions workflow files here in the gitflow repo

Reply:
In line 80 I already referenced the file, or do you want separate links with exact line numbers as well?

Reply:
Thats fine